sdks/python/apache_beam/examples/inference/README.md

```diff
@@ -0,0 +1,114 @@
+PyTorch implementation of RunInference, and then writes the predictions
+to a text file in GCS.
+
+### Data
+Data related to RunInference has been staged in
+`gs://apache-beam-ml/` for use with these example pipelines:
+
+<!---
+Add once benchmark test is released
+- `gs://apache-beam-ml/testing/inputs/it_mobilenetv2_imagenet_validation_inputs.txt`:
+  text file containing the GCS paths of the images of all 5000 imagenet validation data
+    - gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00000001.JPEG
+    - ...
+    - gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00050000.JPEG
+-->
+- `gs://apache-beam-ml/testing/inputs/imagenet_validation_inputs.txt/`:
+  text file containing the GCS paths of the images of a subset of 15 imagenet
+  validation data
+    - gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00000001.JPEG
+    - ...
+    - gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00000015.JPEG
+
+- `gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_*.JPEG`:
+  JPEG images for the entire validation dataset.
+
+- `gs://apache-beam-ml/models/imagenet_classification_mobilenet_v2.pt`: Path to
+  the location of the saved state_dict of the pretrained mobilenet_v2 model
+  from the `torchvision.models` subdirectory.
+
+### Running `pytorch_image_classification.py`
+
+To run the image classification pipeline locally, use the following command:
+```sh
+python -m apache_beam.examples.inference.pytorch_image_classification \
+  --input gs://apache-beam-ml/testing/inputs/it_mobilenetv2_imagenet_validation_inputs.txt \
+  --output predictions.csv \
+  --model_state_dict_path gs://apache-beam-ml/models/imagenet_classification_mobilenet_v2.pt
+```
+
+This will write the output to the `predictions.csv` with contents like:
+```
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005002.JPEG,333
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005003.JPEG,711
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005004.JPEG,286
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005005.JPEG,433
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005006.JPEG,290
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005007.JPEG,890
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005008.JPEG,592
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005009.JPEG,406
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005010.JPEG,996
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005011.JPEG,327
+gs://apache-beam-ml/datasets/imagenet/raw-data/validation/ILSVRC2012_val_00005012.JPEG,573
+```
+where the second item in each line is the integer representing the predicted class of the
+image.
```

Comment:
it would be cool if one of the ptransforms in the example joined to integer prediction to the actual name of the image.

for example:
gs://apache-beam-ml/datasets/.....5102.jpeg, horse
gs://apache-beam-ml/datasets/.....5102.jpeg, cheese

etc.

But that is outside of the scope of this PR.

Reply:
I have that for a different example https://github.com/apache/beam/pull/21766